### ISSUE TITLE
Make it possible to define JS functions as Tcl procs

### ISSUE THREAD

**Issue by uditha-atukorala** [nukedzn/node-tcl#8]
This was asked in [issue 1](https://github.com/nukedzn/node-tcl/issues/1#issuecomment-122303245) and would be a handy thing to have. 

e.g

``` js
// javascript function reference
var multiply = function ( x, y ) {
    return (x * y);
};

// define a Tcl proc 
tcl.proc( 'multiply', multiply );

// usage
tcl.$.set( 'x', '[multiply 10 20]' );
```
#### Discussion points
- How would this look like for the asynchronous interfaces (`eval()` and `queue()`)?

**Comment by ekarak** [comment]
should also target anonymous functions, such as:
```
tcl.proc('multiply', function(x,y) {
    return (x*y);
});
```

**Comment by ekarak** [comment]
One advanced concept I'd find interesting would be to couple Tcl's event loop with ES6 promises.
Thus, heavy/time consuming/external system JS calls functions returning Promises would yield control immediately back to Tcl's interpreter and maybe fire vwait's on the Promise result

**Comment by ekarak** [comment]
OK, starting work on this beast of a feature. I find this idiom a bit redundant though:
`tcl.proc('multiply', multiply)`
How bout
`tcl.expose(multiply)`
or 
`tcl.expose(multiply).as('multiply')`

**Comment by uditha-atukorala** [comment]
I wouldn't mind `tcl.expose(multiply).as('multiply')` and agree it looks nicer :smile:  (not sure how `tcl.expose(multiply)` would work though).

**Comment by uditha-atukorala** [comment]
@ekarak, if you are going to work on this please assign it to yourself.

**Comment by ekarak** [comment]
I can't :) need permissions probably.
I've forked and I'm working on my own repo anyway.

**Comment by uditha-atukorala** [comment]
Sorted...

**Comment by uditha-atukorala** [comment]
@ekarak, are you still working on this?

**Comment by uditha-atukorala** [comment]
Implemented in #15
